addon/mixins/data-adapter-mixin.js

```diff
@@ -94,8 +95,10 @@ export default Ember.Mixin.create({
   */
   handleResponse(status) {
     if (status === 401) {
-      this.get('session').invalidate();
-      return true;
+      if (this.get('session.isAuthenticated')) {
+        this.get('session').invalidate();
+      }
+      return new DS.InvalidError(['Session is not authenticated']);
```

Comment:
ah, ok.  I can revert that then, and can be addressed as a separate branch if needed.

The error in my case (#721) is related to session invalidation anyways.